sensor/openvas.go

```diff
@@ -0,0 +1,173 @@
+package sensor
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+	"sync"
+)
+
+var processes = make(map[string]*os.Process)
```

Comment:
Maybe make an interface, or something equivalent and move all the `addProcess`, `deleteProcess` into that interface, instead of an global `processes` variable that is called in an hardcoded manner in these functions?

Reply:
I created a struct with two methods (`addProcess` and `deleteProcess`). The only thing remaining is the global process list but the only solution in my mind is to create the process list in the scheduler and it passes it to the openvas functions. It would be a lot of passing.